build/relsignDSCModules.sh

```diff
@@ -0,0 +1,309 @@
+#!/bin/bash
+
+# Script to sign DSC modules and drop off signed packages (in zip format) into build share.
+# It generates a checksum file for each DSC module, uploads to azure storage account using SAS_SOURCE_URI,
+# then downloads the signed *.asc file from SAS_DEST_URI and 
+# serializes it into the zip package for given DSC module.
+
+StorageVersion="2014-02-14"
+# Max number of attempts to wait for signing worklow to finish 
+MaxSignWaitAttempts=30
+
+# read_blob <URI> <REMOTE_FILE> <DEST_FILE>
+read_blob() {
+
+    [ ! -z "$1" ] || error "read_blob(): expected remote URI" 1
+    [ ! -z "$2" ] || error "read_blob(): expected remote filename" 1
+    [ ! -z "$3" ] || error "read_blob(): expected destination file" 1
+
+    url="$1"
+    remotefile="$2"
+    localfile="$3"
+
+    container=$( echo "$1" | awk -F \?  '{container=index($0,"?"); print $1}' )
+	echo "container is $container"
+    sas=$( echo "$url" | awk -F \? '{container=index($0,"?"); print $2}' )
+    url="${container}/${remotefile}?${sas}"
+
+    timestamp=$(date +%Y-%m-%dT%H:%M:%SZ)
+    output=$(curl --silent --write-out %{http_code} --request "GET" \
+                  --header "x-ms-date    : $timestamp" \
+                  --header "x-ms-version : $StorageVersion" \
+                  "$url" )
```

Comment:
I don't understand the protocol, so I won't be reviewing this for "correctness" in signing - I'll leave that to someone else more familiar.